File: app/src/main/java/com/jlrutilities/subnetapp/adapters/CheatsheetArrayAdapter.java
package com.jlrutilities.subnetapp.adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import com.jlrutilities.subnetapp.R;

public class CheatsheetArrayAdapter extends ArrayAdapter<String> {
  private final Context context;
  private final String[] bitsArr;
  private final String[] netmaskArr;
  private final String[] hostsArr;

  public CheatsheetArrayAdapter(Context context, String[] values, String[] netmaskArr, String[] hostsArr) {
    super(context, -1, values);
    this.context = context;
    this.bitsArr = values;
    this.netmaskArr = netmaskArr;
    this.hostsArr = hostsArr;
  }

  @Override
  public View getView(int position, View convertView, ViewGroup parent) {
    LayoutInflater inflater = (LayoutInflater) context
        .getSystemService(Context.LAYOUT_INFLATER_SERVICE);
    View rowView = inflater.inflate(R.layout.cheatsheet_list_item, parent, false);

    TextView bitsTv = rowView.findViewById(R.id.bitsItemTv);
    TextView netmaskTv = rowView.findViewById(R.id.netmaskItemTv);
    TextView hostsTv = rowView.findViewById(R.id.hostsItemTv);

    bitsTv.setText(bitsArr[position]);
    netmaskTv.setText(netmaskArr[position]);
    hostsTv.setText(hostsArr[position]);

    return rowView;
  }
}

File: app/src/main/java/com/jlrutilities/subnetapp/adapters/CheatsheetArrayAdapter.kt
package com.jlrutilities.subnetapp.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.TextView
import com.jlrutilities.subnetapp.R


//** Allows subnet context to be passed and reference view information. */
class CheatsheetArrayAdapter(
        context: Context,
        values: Array<String>,
        netmaskArr: Array<String>,
        hostsArr: Array<String>
): ArrayAdapter<String>(context, -1, values){

  private val context: Context
  private val bitsArr: Array<String>
  private val netmaskArr: Array<String>
  private val hostsArr: Array<String>

  init {
    this.context = context;
    this.bitsArr = values;
    this.netmaskArr = netmaskArr;
    this.hostsArr = hostsArr;
  }


  //** Sets subnet information in view. */
  override fun getView(
          position: Int,
          convertView: View?,
          parent: ViewGroup
  ): View {
    val inflater: LayoutInflater = LayoutInflater.from(context)
    // TODO: ViewHolder Pattern
    val rowView: View = inflater.inflate(R.layout.cheatsheet_list_item, parent, false);

    val bitsTv: TextView = rowView.findViewById(R.id.bitsItemTv);
    val netmaskTv: TextView = rowView.findViewById(R.id.netmaskItemTv);
    val hostsTv: TextView = rowView.findViewById(R.id.hostsItemTv);

    bitsTv.text = bitsArr[position]
    netmaskTv.text = netmaskArr[position]
    hostsTv.text = hostsArr[position]

    return rowView;
  }
}
